src/main/java/org/opentripplanner/middleware/utils/TemplateUtils.java

```diff
@@ -0,0 +1,61 @@
+package org.opentripplanner.middleware.utils;
+
+import freemarker.template.Configuration;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateModelException;
+import org.opentripplanner.middleware.bugsnag.BugsnagReporter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.List;
+
+/**
+ * A util class that helps with rendering templates
+ */
+public class TemplateUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(TemplateUtils.class);
+
+    private static final Configuration config = new Configuration(Configuration.VERSION_2_3_30);
+    private static final String BASE_TEMPLATE_PATH = "/templates/";
+    private static final List<String> sharedConfigKeys = List.of(
+        "OTP_ADMIN_DASHBOARD_NAME", "OTP_ADMIN_DASHBOARD_URL", "OTP_UI_NAME", "OTP_UI_URL"
+    );
+
+    /**
+     * Initializes the templating engine with various configurations. This must be called after
+     * {@link ConfigUtils#loadConfig} is called.
+     */
+    public static void initialize() {
+        try {
+            config.setClassForTemplateLoading(TemplateUtils.class, BASE_TEMPLATE_PATH);
+            config.setDefaultEncoding("UTF-8");
+            for (String key : sharedConfigKeys) {
+                config.setSharedVariable(key, ConfigUtils.getConfigPropertyAsText(key));
+            }
+        } catch (TemplateModelException e) {
+            LOG.error("An error occurred while initializing FreeMarker: ", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Renders a template given an object.
+     * @param templatePath  path to template file (.ftl that is found in the {@link #BASE_TEMPLATE_PATH} resources
+     *                     directory
+     * @param data          template data (any kind of public class that has public getXxx/isXxx methods as prescribed
```

Comment:
Suggestion: rename to `templateFields`. Also, I think we should restrict the type to `Map<String, Object>`, that is the only type passed to this method so far. We can add an overload method if other needs arise.

Reply:
I made this change previously and @evansiroky had some issues with it, so in the interest of moving forward, I think I'll leave as is.